src/tools/build_helper/src/git.rs

```diff
@@ -108,6 +105,37 @@ pub fn get_git_merge_base(
     Ok(output_result(git.arg("merge-base").arg(&updated_master).arg("HEAD"))?.trim().to_owned())
 }
 
+/// Resolves the closest merge commit by the given `author` and `target_paths`.
+///
+/// If it fails to find the commit from upstream using `git merge-base`, fallbacks to HEAD.
+pub fn get_closest_merge_commit(
```

Comment:
Could you add a comment explaining how does this differ from just using merge base (I know that it is to handle missing upstream remotes, but currently the information is only in a commit message, would be nicer to be in a comment).

Reply:
I am not sure if explaining `get_git_merge_base` in the `get_closest_merge_commit` doc-comments would be a good idea as `get_git_merge_base` is a private function. I just pushed https://github.com/rust-lang/rust/pull/130161/commits/991a60dc7076d550921121d95d861cdc8f16216b, let me know if you like that approach.

Reply:
Thank you! I think it's clearer now.